software/lib/snap.c

```diff
@@ -772,7 +773,8 @@ int snap_action_completed(struct snap_action *action, int *rc, int timeout)
 	int _rc = 0;
 	uint32_t action_data = 0;
 	struct snap_card *card = (struct snap_card *)action;
-	int t0, dt, timeout_ms;
+	unsigned long t0;
+	int dt, timeout_ms;
```

Comment:
Why 32-bit? Use 64-bit here too. Such that you do not need to cast the calculation below, giving still a little unsure feeling that it is now right even if the time difference is rediciously large.

Reply:
dt is the time diff in msec. This will never wrap !